pymc_marketing/mlflow.py

```diff
@@ -412,14 +460,319 @@ def log_inference_data(
+    ) -> Any:
+        """Perform predictions or sampling using the specified prediction method.
+
+        Parameters
+        ----------
+        context : Any
+            The context in which the model is running. Isn't specified by users but is passed by MLflow.
+        model_input : array, shape (n_pred, n_features)
+            The input data used for prediction.
+        params : dict, optional
+            A dictionary of parameters to specify the prediction method.
+
+        Returns
+        -------
+        ndarray or InferenceData
+            The predictions or samples generated by the model.
+
+        Raises
+        ------
+        ValueError
+            If an unsupported prediction method is specified.
+
+        """
+        # Use the class-level predict_method if params is not provided or doesn't contain 'predict_method'
+        params = params or {"predict_method": "predict"}
+        predict_method = params.get("predict_method", self.predict_method)
+
+        if predict_method == "predict":
+            return self.model.predict(
+                model_input,
+                extend_idata=self.extend_idata,
+                include_last_observations=self.include_last_observations,
+                original_scale=self.original_scale,
+                var_names=self.var_names,
+                **self.sample_kwargs,  # type: ignore[arg-type]
+            )
+        elif predict_method == "sample_posterior_predictive":
+            return self.model.sample_posterior_predictive(
+                model_input,
+                extend_idata=self.extend_idata,
+                combined=self.combined,
+                include_last_observations=self.include_last_observations,
+                original_scale=self.original_scale,
+                var_names=self.var_names,
+                **self.sample_kwargs,  # type: ignore[arg-type]
+            )
+        elif predict_method == "sample_prior_predictive":
+            return self.model.sample_prior_predictive(
+                model_input,
+                extend_idata=self.extend_idata,
+                combined=self.combined,
+                var_names=self.var_names,
+                **self.sample_kwargs,  # type: ignore[arg-type]
+            )
+        else:
+            raise ValueError(
+                f"The prediction method '{predict_method}' is not supported."
+            )
+
+
+def log_model(
+    mmm: MMM,
+    artifact_path: str = "model",
+    mmm_registry_name: str | None = None,
+    **wrapper_kwargs,
+) -> None:
+    """Log a PyMC-Marketing MMM as an MLflow artifact for the current run.
+
+    Parameters
+    ----------
+    model : MMMWrapper
+        The MMM to be logged.
+    artifact_path : str, optional
+        The path to the artifact to be logged. Defaults to "mmm_model".
+    conda_env : dict, optional
+        A dictionary representation of a Conda environment. Defaults to the default conda environment.
+    registered_model_name : str, optional
```

Comment:
doesn't match up with the parameter name. I prefer this name over one with a prefix of "mmm"